app/src/backend/ActivityService.ts

```diff
@@ -21,9 +21,241 @@ import type { Transaction } from "sequelize";
 
 type GasValueInput = Omit<GasValueCreationAttributes, "id"> & {
   emissionsFactor?: Omit<EmissionsFactorAttributes, "id">;
+  id?: string;
```

Comment:
```suggestion
type GasValueInput = GasValueCreationAttributes & {
  emissionsFactor?: EmissionsFactorAttributes;
```

No need to omit id and then add it again.

Reply:
And I guess you might want to add id for the emissions factor so it knows which one to update.
Not sure if id needs to be optional for EmissionsFactor so we can also create new ones or if that always already happened in the POST endpoint and we can just assume they are present in the PATCH endpoint.

Reply:
Original comment has not been resolved